parser/src/python.lalrpop

```diff
@@ -1603,21 +1598,32 @@ FunctionArgument: (Option<(TextSize, TextSize, Option<ast::Identifier>)>, ast::E
     <location:@L> "**" <e:Test<"all">> <end_location:@R> => (Some((location, end_location, None)), e),
 };
 
+/// Comma separated sequence that allows an optional trailing comma.
 #[inline]
 Comma<T>: Vec<T> = {
-    <items: (<T> ",")*> <last: T?> => {
-        let mut items = items;
-        items.extend(last);
-        items
+    <mut v:(<T> ",")*> <last:T?> => {
+        if let Some(element) = last {
+            v.push(element);
+        }
+        v
     }
 };
 
-#[inline]
+/// One ore more items that are separated by a comma.
 OneOrMore<T>: Vec<T> = {
-    <i1: T> <i2:("," T)*> => {
-        let mut items = vec![i1];
-        items.extend(i2.into_iter().map(|e| e.1));
-        items
+    <e:T> => vec![e],
+    <mut v: OneOrMore<T>> "," <e:T> => {
+        v.push(e);
+        v
+    }
+};
```

Comment:
This should be the same as 

```
#[inline]
OneOrMore<T>: Vec<T> = {
	<mut v: (<T> ",")*> <last:T> => {
		v.push(last);
		v
	}
}
```

But this doesn't work for reasons I do not understand. I don't know if both macros will expand to the same code or not.